src/Dfc.CourseDirectory.Functions/FixCorruptStatusData.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.Core.Models;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace Dfc.CourseDirectory.Functions
+{
+    /// <summary>
+    /// In prod CourseStatus has got out of sync with the CourseRuns
+    /// </summary>
+    public class FixCorruptStatusData
+    {
+        private readonly Configuration _configuration;
+        private readonly ILogger _logger;
+        private readonly DocumentClient _documentClient;
+
+        public FixCorruptStatusData(
+            Configuration configuration,
+            ILoggerFactory loggerFactory,
+            DocumentClient documentClient)
+        {
+            _configuration = configuration;
+            _documentClient = documentClient;
+            _logger = loggerFactory.CreateLogger<FixCorruptStatusData>();
+        }
+
+        [FunctionName(nameof(FixCorruptStatusData))]
+        [NoAutomaticTrigger]
+        public async Task Run(string input)
+        {
+            var collectionUri = UriFactory.CreateDocumentCollectionUri(
+                _configuration.DatabaseId,
+                _configuration.CoursesCollectionName);
+
+            const CourseStatus courseRunStatus = CourseStatus.Live;
+            const CourseStatus unwantedCourseStatus = CourseStatus.BulkUploadReadyToGoLive;
+
+            var query = _documentClient
+                .CreateDocumentQuery<Course>(collectionUri,
+                    new FeedOptions() {EnableCrossPartitionQuery = true})
+                .Where(c => c.CourseStatus == unwantedCourseStatus)
+                .AsDocumentQuery();
+
+            int fixCount = 0;
+            await query.ProcessAll(async courses =>
```

Comment:
Does this need to be done in a `ProcessAll`? If you enumerate `query` you can just iterate using a `foreach`?